File: lista_2/bancoimobiliario.py
grana, jogos = map(int, input().split())
D = grana
E = grana
F = grana
for i in range(jogos):
    capitalismo = list(map(str, input().split()))
    capitalismo[-1] = int(capitalismo[-1])
    if capitalismo[0] == 'C':
        if capitalismo[1] == 'D':
            D = D - capitalismo[2]
        elif capitalismo[1] == 'E':
            E = E - capitalismo[2]  
        elif capitalismo[1] == 'F':
            F = F - capitalismo[2]
            
    elif capitalismo[0] == 'V':
        if capitalismo[1] == 'D':
            D = D + capitalismo[2]
        elif capitalismo[1] == 'E':
            E = E + capitalismo[2]
        elif capitalismo[1] == 'F':
            F = F + capitalismo[2]
            
    elif capitalismo[0] == 'A':
        if capitalismo[1] == 'D':
            if capitalismo[2] == 'E':
                D = D + capitalismo[3]
                E = E - capitalismo[3]
            elif capitalismo[2] == 'F':
                D = D + capitalismo[3]
                F = F - capitalismo[3]
        elif capitalismo[1] == 'E':
            if capitalismo[2] == 'D':
                E = E + capitalismo[3]
                D = D - capitalismo[3]
            elif capitalismo[2] == 'F':
                E = E + capitalismo[3]
                F = F - capitalismo[3]
        elif capitalismo[1] == 'F':
            if capitalismo[2] == 'D':
                F = F + capitalismo[3]
                D = D - capitalismo[3]
            elif capitalismo[2] == 'E':
                F = F + capitalismo[3]
                E = E - capitalismo[3]
                
print(D, E, F)

File: lista_2/BancoImobiliario.java
package beecrowd_Java\lista_2;

import java.io.IOException;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		
		Scanner input = new Scanner(System.in);
		List<Integer> configuracao = new ArrayList<>();
		
		for (int i = 0; i < 2; i++) {
			configuracao.add(input.nextInt());
		}
		
		int saldo = configuracao.get(0);
		int rodadas = configuracao.get(1);
		int d = saldo;
		int e = saldo;
		int f = saldo;
		
		for (int k = 0; k < rodadas; k++) {
			
			String[] acoes = input.nextLine().split(" ");
			
			String acao  = acoes[0];
			String pessoa1 = acoes[1];
			String pessoa2 = "N";
			int reais = Integer.parseInt(acoes[2]);
			
			if (pessoa1 == "D") {
				
				if (acao == "C") {
					d -= reais;
					
				} else if (acao == "V") {
					d += reais;
					
				} else if (acao == "A") {
					reais = Integer.parseInt(acoes[3]);
					pessoa2 = acoes[2];
					d += reais;
					if (pessoa2 == "E") {
						e -= reais;
					} else if (pessoa2 == "F") {
						f -= reais;
					}
				}
				
			} else if (pessoa1 == "E") {
				
				if (acao == "C") {
					e -= reais;
					
				} else if (acao == "V") {
					e += reais;
					
				} else if (acao == "A") {
					reais = Integer.parseInt(acoes[3]);
					pessoa2 = acoes[2];
					e += reais;
					if (pessoa2 == "D") {
						d -= reais;
					} else if (pessoa2 == "F") {
						f -= reais;
					}
				}
				
			} else if (pessoa1 == "F") {
				
				if (acao == "C") {
					f -= reais;
					
				} else if (acao == "V") {
					f += reais;
					
				} else if (acao == "A") {
					reais = Integer.parseInt(acoes[3]);
					pessoa2 = acoes[2];
					f += reais;
					if (pessoa2 == "D") {
						d -= reais;
					} else if (pessoa2 == "E") {
						e -= reais;
					}
				}
				
			} 
			
			System.out.println(d + " " + e + " " + f);
		}

	}

}